public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -100,35 +102,29 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
     });
   }
 
-  private _updateDirty = debounce(() => {
-    const diff = jsonDiff(vizPanelToPanel(this.state.sourcePanel.resolve()), vizPanelToPanel(this.state.panel));
-    const diffCount = Object.values(diff).reduce((acc, cur) => acc + cur.length, 0);
-    this.setState({ isDirty: diffCount > 0 });
+  private _onActivate() {
+    this.loadDataSource();
+    const changesSub = this.subscribeToEvent(SceneObjectStateChangedEvent, this._handleStateChange);
```

Comment:
Since the events bubble up, we don't need to subscribe to any children, we can subscribe to the VizPanelManager